### PR TITLE
feat: Finish Screen for Payment Link Sharing

### PR DESCRIPTION
**Pull request by CtpN3m01**
# 📝 Summary
This PR finalizes the screen for Payment Link Sharing, enhancing the `PaymentLinkShare` component with new functionality, UI improvements, and responsive design. While it fixes copy-related issues and cleans up unused code and dependencies, some changes were made outside the original scope of the issue.

---

## 🔗 Related Issues
- Closes #20

---

#### Basic Component Structure
- Created the `PaymentLinkShare` component with an initial container and the title: *"Now share your payment link as you prefer."*

#### Copy to Clipboard Functionality
- Added `linkCopyMessage` and `embedCopyMessage` states to manage feedback messages.
- Implemented the `copyToClipboard` function for both payment links and embed codes.

#### Embed Code Section
- Added a section for copying embed codes with a confirmation message.

#### Social Sharing Section
- Updated the Twitter icon to reflect the new logo using FontAwesome.
- Installed `@fortawesome/react-fontawesome` and `@fortawesome/free-brands-svg-icons` to support the updated icon.

#### Responsive Design
- Enhanced input fields for better usability and readability.
- Implemented responsive layouts using Tailwind CSS for seamless viewing on all screen sizes.

#### Input Behavior Enhancement
- Disabled the default black selection border for the Payment Link input field to improve the visual design.

---

#### Currency Selector
- Added a currency selector dropdown with flag icons (USD, EUR, GBP).
- This feature was not part of the original issue but provides additional functionality for currency management.

#### Client Menu
- Created a client menu with options like View Profile, Settings, Notifications, and Logout.
- This menu, while useful, was outside the defined scope.

#### Removed Header Section and Ariakit Package
- These elements were initially added while working on unrelated functionality but were removed to align with the scope.

---

## 🖼️ Current Output

https://github.com/user-attachments/assets/f9e4295e-a289-445e-954c-cf1322e344c9

Captura de pantalla 2024-11-25 120341

---

### Cross-Device Testing
- Verified on desktop, mobile, and various browsers (Chrome, Safari, Firefox).
- Ensured the copy-to-clipboard functionality works seamlessly with fallback for older browsers.

### UI Validation
- Tested all components for responsiveness across different screen sizes.
- Confirmed smooth animations and proper rendering of icons and buttons.
- Verified the updated Twitter icon in the Social Sharing Section.
- Validated the Payment Link input field to ensure the default black selection border was disabled.

### Functional Tests
- Checked the social sharing buttons' functionality for each platform.
- Verified feedback messages for copying links and embed codes.

---

## 💬 Comments
The Currency Selector and Client Menu were added as enhancements but later removed to align with the issue's scope. Their inclusion in the commits is acknowledged, but they have been excluded from the final implementation. Please let me know if additional adjustments are necessary.

### CODE REVIEW

File: paystell-frontend/src/app/payment-link/page.tsx [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+              className="w-full md:w-3/5 border px-4 py-2 rounded-lg focus:outline-none"
+              style={{ userSelect: 'none' }}
+            />
+            <button
+              onClick={() => copyToClipboard(Link, setLinkCopyMessage)}
+              className="w-full md:w-auto bg-[#009EFF] text-white px-4 py-2 rounded-lg flex items-center justify-center space-x-2 transition-transform transform hover:scale-105 active:scale-95"
+            >
+              <Copy className="w-4 h-4" />
+              <span>Copy Link</span>
+            </button>
```
Comment (MPSxDev) [inline paystell-frontend/src/app/payment-link/page.tsx:75]: please use Button component located in components/ui for all buttons 
```suggestion
              <Button
                type="button"
                variant="default"
                onClick={() => copyToClipboard(Link, setLinkCopyMessage)}
              >
              <Copy className="w-4 h-4" />
              Copy Link
              </Button>
            
```

File: paystell-frontend/src/app/payment-link/page.tsx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+          <h1 className="text-xl font-bold mb-4" style={{ fontSize: '24px' }}>Now share your payment link as you prefer</h1>
+          <div className="mb-4 flex flex-col md:flex-row items-start md:items-center space-y-2 md:space-y-0 md:space-x-2">
+            <input
+              type="text"
+              value={Link}
+              readOnly
+              className="w-full md:w-3/5 border px-4 py-2 rounded-lg focus:outline-none"
+              style={{ userSelect: 'none' }}
+            />
+            <button
```
Comment (MPSxDev) [inline paystell-frontend/src/app/payment-link/page.tsx:69]: use input component located in components/ui

File: paystell-frontend/src/app/payment-link/page.tsx [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+              <span>Twitter</span>
+            </button>
+            <button className="w-full md:w-auto bg-green-500 text-white px-4 py-2 rounded-lg flex items-center justify-center space-x-2 transition-transform transform hover:scale-105 active:scale-95">
+              <FaWhatsapp className="w-4 h-4" />
+              <span>WhatsApp</span>
+            </button>
+            <button className="w-full md:w-auto bg-blue-700 text-white px-4 py-2 rounded-lg flex items-center justify-center space-x-2 transition-transform transform hover:scale-105 active:scale-95">
+              <FaLinkedin className="w-4 h-4" />
+              <span>LinkedIn</span>
+            </button>
```
Comment (MPSxDev) [inline paystell-frontend/src/app/payment-link/page.tsx:125]: same for the Button component/ui

File: paystell-frontend/src/app/payment-link/page.tsx [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+              rows={4}
+            />
+            <div className="flex items-center w-full md:w-auto">
+              <button
+                onClick={() => copyToClipboard(embedCode, setEmbedCopyMessage)}
+                className="w-full md:w-auto bg-[#009EFF] text-white px-4 py-2 rounded-lg flex items-center justify-center space-x-2 transition-transform transform hover:scale-105 active:scale-95"
+              >
+                <Code className="w-4 h-4" />
+                <span>Copy Embed Code</span>
+              </button>
```
Comment (MPSxDev) [inline paystell-frontend/src/app/payment-link/page.tsx:100]: same for the Button component/ui

File: paystell-frontend/src/app/payment-link/page.tsx
Comment (MPSxDev) [inline paystell-frontend/src/app/payment-link/page.tsx:1]: move to a separated component in `components/dashboard/links/ShareLink.tsx` follow the example of app/dashboard/links

**Review by MPSxDev — CHANGES_REQUESTED** [review]

File: paystell-frontend/src/app/payment-link/page.tsx
Comment (MPSxDev) [reply paystell-frontend/src/app/payment-link/page.tsx:1]: rigth!

File: paystell-frontend/src/app/ShareLink/page.tsx
[the remark this answers is not in the document]
Comment (CtpN3m01) [reply paystell-frontend/src/app/ShareLink/page.tsx:1]: image

Ready

**Review by MPSxDev — APPROVED** [review]
excellent thanks

**Pull request merged**